platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/MarkerViewManager.java

```diff
@@ -503,18 +504,24 @@ public void invalidateViewMarkersInVisibleRegion() {
                 }
               }
 
-              adaptedView.setOnClickListener(new View.OnClickListener() {
-                @Override
-                public void onClick(final View v) {
-                  boolean clickHandled = false;
-                  if (onMarkerViewClickListener != null) {
-                    clickHandled = onMarkerViewClickListener.onMarkerClick(marker, v, adapter);
-                  }
+              adaptedView.setOnTouchListener(new View.OnTouchListener() {
 
-                  if (!clickHandled) {
-                    ensureInfoWindowOffset(marker);
-                    select(marker, v, adapter);
+                @Override
+                public boolean onTouch(View v, MotionEvent event) {
+                  switch (event.getAction()) {
```

Comment:
Why using `switch` instead `if`? There is only one case (`MotionEvent.ACTION_UP`).